lib/game.rb

```diff
@@ -0,0 +1,63 @@
+require 'pry'
+
+class Game
+
+  WIN_COMBINATIONS = [
+  [0,1,2],  # Top row
+  [3,4,5],  # Middle row
+  [6,7,8],  # Bottom row
+  [0,3,6],  # Vertical row
+  [1,4,7],  # Vertical Middle row
+  [2,5,8],  # Vertical row
+  [0,4,8],  # Diagonal Row
+  [6,4,2]   # Diagonal Row
+  ]
+
+  attr_accessor :board, :player_1, :player_2
+
+  def initialize(player_1=Human.new("X"), player_2=Human.new("O"), board=Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end 
+
+  def current_player
+    board.turn_count.even? ? player_1 : player_2
+  end
+
+  def over?
+    board.full? || won? || draw?
+  end 
+
+  def won?
+    WIN_COMBINATIONS.detect do |combo| 
+    combo.all? {|location| board.cells[location] == "X" } || combo.all? {|location| board.cells[location] == "O" }
+    end
+  end 
+
+  def draw?
+    board.full? && !won?
+  end
+
+  def winner
+    won? ? board.position(won?[0] + 1) : nil 
+  end  
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.